src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -196,10 +218,13 @@ public boolean mkdirs(Path f, FsPermission permission) throws IOException {
      */
     @Override
     public FileStatus getFileStatus(Path f) throws IOException {
-//        Path mappedPath = pathMapper.mapPath(f);
-//        Configuration conf = getConf();
-//        FileSystem fs = mappedPath.getFileSystem(conf);
-//        return fs.getFileStatus(mappedPath);
-        return null;
+        FileSystemPathTuple tuple = generateFSPathTuple(f);
+        return tuple.getFileSystem().getFileStatus(f);
+    }
+
+    private FileSystemPathTuple generateFSPathTuple(Path p) throws IOException {
+        Path mappedPath = this.pathMapper.mapPath(p);
+        FileSystem fs = mappedPath.getFileSystem(getConf());
+        return new FileSystemPathTuple(fs, mappedPath);
```

Comment:
Thinking about it: I don't understand how this works after the current working directory is changed.  If I look at `"./path/to/obj"`, we will not know **which** underlying filesystem to open, so it doesn't matter that I changed directories in that FS.